dspace/modules/api/src/main/java/org/dspace/submit/step/SelectPublicationStep.java

```diff
@@ -308,14 +151,108 @@ private void addEmailsAndEmbargoSettings(DryadJournalConcept journalConcept, Ite
         }
     }
 
-    private boolean processJournal(DryadJournalConcept journalConcept, String manuscriptNumber, Item item, Context context,
-                                   HttpServletRequest request, String articleStatus) throws AuthorizeException, SQLException {
+    private boolean processJournal(Item item, Context context, HttpServletRequest request) throws AuthorizeException, SQLException {
+        String articleStatus = request.getParameter("article_status");
+        // if article status is ACCEPTED, user enters journal name and manuscript number
+        // if article status is PUBLISHED, user can either enter journal name or pub DOI
+        // if article status is ARTICLE_STATUS_IN_REVIEW, user chooses journal from pulldown and optionally enters manuscript number
+
+        if (articleStatus==null) {
+            EventLogger.log(context, "submission-select-publication", "error=exception_reselect_journal");
+            return false;
+        }
+
+        // Look for a DOI or PMID entered in the UI
+        // Then use crosswalk to pre-load metadata into the Item.
+        String identifier = request.getParameter("article_doi");
+        if (identifier!=null && !identifier.equals("")) {
+            // normalize and validate the identifier
+            Matcher doiMatcher = Pattern.compile("(doi:)*(.+\\/.+)").matcher(identifier);
```

Comment:
According to the DOI spec, a DOI just has to have a prefix and a suffix separated by a slash, so even though most are 10.something slash something, I don't know that they have to be. Certainly I don't think there's any harm in asking crossref to look up bad ones.